gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef GAZEBO_ROS_NODE_HPP
+#define GAZEBO_ROS_NODE_HPP
+
+#include <memory>
+#include <mutex>
+#include <atomic>
+
+#include <rclcpp/rclcpp.hpp>
+#include <gazebo/common/common.hh>
+
+#include <gazebo_ros/executor.hpp>
+
+namespace gazebo_ros
+{
+
+/// ROS Node for gazebo plugins
+class Node : public rclcpp::Node
+{
+public:
+  /// Exception thrown when a #Load is called for a Node before #InitRos has ever been called
+  class NotInitializedException : public std::runtime_error
+  {
+    public:
+      NotInitializedException();
+  };
+
+  typedef std::shared_ptr<Node> SharedPtr;
+
+  virtual ~Node();
+
+  /**
+   * Contruct #node_ and add it to the global #executor_.
+   * \param[in] node_name Name of node to create
+   */
+  static SharedPtr Create(const std::string& node_name);
+
+  /**
```

Comment:
I realize the ROS 1 code mixes `///` with `/**`, the [ROS 2 developer guide](https://github.com/ros2/ros2/wiki/Developer-Guide#comments-and-doc-comments) accepts either and they're both [supported by Doxygen](https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html#cppblock).

I think there's value in being consistent within a single repository though, so I think we should define now which one to use. I don't have a preference, so I'll leave up to you or anyone with strong opinions.

Reply:
I agree it should be consistent. Looks like some other [ros2 code](https://github.com/ros2/rcl/blob/master/rcl/include/rcl/client.h) always starts with ```///``` and adds ```/** */``` if more lines are required

Reply:
Oh interesting, that looks weird to me on a first look, I wonder how that gets rendered and what's the rationale behind it.